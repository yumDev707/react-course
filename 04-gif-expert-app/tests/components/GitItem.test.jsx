const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe('Testing GifItem component', () => {
    const title = 'Dragon Ball';
    const url = 'http://test.es/';

    test('Eval snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Should show the image with the given url and alt', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} =screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Should show the title', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})