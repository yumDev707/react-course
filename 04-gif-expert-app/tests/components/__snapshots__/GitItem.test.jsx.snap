// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem component Eval snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="http://test.es/"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
